packages/simulation/pitop/simulation/sprites.py

```diff
@@ -85,3 +102,421 @@ def handle_event(type, target_name, component):
 
         elif type == pygame.MOUSEBUTTONUP:
             component.pin.drive_high()
+
+
+class Buzzer(pygame.sprite.Sprite, SimSprite):
+    Size = (int((PMA_CUBE_SIZE[0] + 2 * MARGIN) * 2), PMA_CUBE_SIZE[0] + 2 * MARGIN)
+
+    def __init__(self, config, scale):
+        pygame.sprite.Sprite.__init__(self)
+
+        def get_pos(image):
+            return (
+                int((self.Size[0] - image.get_width()) / 2),
+                int((self.Size[1] - image.get_height()) / 2),
+            )
+
+        self.component_image = SimSprite._load_image(Images.Buzzer, scale)
+        self.component_pos = get_pos(self.component_image)
+
+        self.buzzer_sound_image = SimSprite._load_image(Images.buzzer_sound_icon, scale)
+        self.buzzer_sound_image_pos = get_pos(self.buzzer_sound_image)
+
+        self.base_image = pygame.Surface(self.Size)
```

Comment:
this is not accounting for the sim scale. when these Size properties are used for the overall sim size they are [scaled](https://github.com/pi-top/pi-top-Python-SDK/blob/e436c14cb53a7d50db511ca8b01e90744ef3903c/packages/simulation/pitop/simulation/simulation.py#L128).

Reply:
it looks like this size is used quite a bit in positioning later so probably we should save the scaled size onto the instance.

eg self.size = [scale * x for x in self.Size]

might want to rename these to class.SimSize and instance.sprite_size or something like that to make it clearer

the Size class properties were mainly for determining the overall window size if this sprite is the main sprite, since in cases like the Pitop this isn't directly related to the size of the pitop sprite itself